Automattic-Tracks-iOS/Crash Logging/SentryEvent+Extensions.swift

```diff
@@ -0,0 +1,33 @@
+import Sentry
+
+extension Event {
+    static func from(
+        error: NSError,
+        level: SentryLevel = .error,
+        user: Sentry.User? = nil,
+        timestamp: Date = Date(),
+        extra: [String: Any] = [:]
+    ) -> Event {
+        let event = Event(level: level)
+
+        if let underlyingError = error.userInfo[NSUnderlyingErrorKey] as? NSError {
+            event.message = SentryMessage(formatted: underlyingError.debugDescription)
+        } else {
+            event.message = SentryMessage(formatted: error.debugDescription)
+        }
+
+        /// The error code/domain can be used to reproduce the original error
+        var mutableExtra = extra
+        mutableExtra["error-code"] = error.code
+        mutableExtra["error-domain"] = error.domain
+
+        error.userInfo.forEach { (key, value) in
+            mutableExtra[key] = value
+        }
+
+        event.timestamp = timestamp
+        event.extra = [String:Any](uniqueKeysWithValues: mutableExtra.sorted { $0.key > $1.key })
```

Comment:
What's the point of sorting the `mutableExtra` by key if you end up putting them in an dictionary, which by definition is a non-ordered collection?

Reply:
Yeah, good point – too much time spent with PHP I guess. Though when I added this it started sorting the entries properly in Sentry. But now I removed it and it does too. Gremlins, I guess?

Removed in dfec0bfa7d0b20e5b679d54cbb2ee9c5e24b1b23